codelabs/custom_renderer/custom_renderer.md

```diff
@@ -1,143 +1,176 @@
 
-## Define and register a renderer
+Before setting up the rest of the application, let's change the storage key used for this codelab application. This will ensure that the workspace is saved in its own storage, separate from the regular sample app, so that we don't interfere with other demos. In `serialization.js`, change the value of `storageKey` to some unique string:
+
+```js
+// Use a unique storage key for this codelab
+const storageKey = 'customRenderersWorkspace';
+```
+
+## Define and register a custom renderer
 
 A **Renderer** is the interface between your custom rendering code and the rest of Blockly. Blockly provides a base renderer with all required fields set to default values.
 
-Your new renderer must extend the base renderer:
+To start, create a new directory at `src/renderers` and add a file inside named `javascript.js`.
+
+At the top of the file, you first need to import `blockly/core`:
+
 ```js
-CustomRenderer = function(name) {
-  CustomRenderer.superClass_.constructor.call(this, name);
-};
-Blockly.utils.object.inherits(CustomRenderer,
-    Blockly.blockRendering.Renderer);
+import * as Blockly from 'blockly/core';
 ```
 
-After defining your renderer you need to tell Blockly that it exists. Register your renderer by name:
+Then define your new custom renderer and have it extend the base renderer:
+
+```js
+class CustomRenderer extends Blockly.blockRendering.Renderer {
+  constructor() {
+    super();
+  }
+}
+```
+
+After defining your renderer you need to tell Blockly that it exists. Register your renderer and give it the name `custom_renderer`:
+
 ```js
 Blockly.blockRendering.register('custom_renderer', CustomRenderer);
 ```
 
-To use your custom renderer, set the `renderer` property in the configuration struct in `playground.html`:
+To use your custom renderer, you first need to import your new file into `src/index.js`.  Add this import at the top of `src/index.js`:
+
 ```js
-Blockly.inject('blocklyDiv', {
-    renderer: 'custom_renderer'
-  }
-);
+import './renderers/javascript';
+```
+
+Now, add the the `renderer` property into the configuration struct passed to `Blockly.inject` so that it now looks like this:
+
+```js
+const ws = Blockly.inject(blocklyDiv, {renderer: 'custom_renderer', toolbox});
 ```
 
 ### The result
 
-To test, open the playground in your browser and drag out a repeat block. The resulting block looks close to normal, but with slightly different alignment than Geras or Thrasos.
+If the server is already running, you can refresh the page to see your changes. Otherwise, run `npm run start` to start the server. Once the server is running, click on the `Loops` entry in your browser and drag out a repeat block. The resulting block will look the same as normal, since your custom renderer has merely extended the default renderer without changing anything.
 
-![](./custom_renderer.png)
+![Screenshot of a renderer with a default appearance.](./custom_renderer.png)
```

Comment:
i'll want to change this alt-text based on how we decide to describe what this now looks like since it's not exactly the same as the default.